loopchain/channel/channel_inner_service.py

```diff
@@ -693,11 +693,27 @@ def get_tx_info(self, tx_hash):
         if last_block is None:
             util.logger.debug("BlockChain has not been initialized yet.")
             return
-        elif self._channel_service.state_machine.state not in ("Vote", "Watch", "LeaderComplain"):
-            util.logger.debug(f"Can't add unconfirmed block in state({self._channel_service.state_machine.state}).")
-            return
 
-        self._channel_service.state_machine.vote(unconfirmed_block=unconfirmed_block)
+        try:
+            self._channel_service.block_manager.verify_confirm_info(unconfirmed_block)
+        except ConfirmInfoInvalid:
+            # TODO
+            pass
+        except ConfirmInfoInvalidNeedBlockSync as e:
+            util.logger.debug(f"ConfirmInfoInvalidNeedBlockSync {e}")
+            block_manager = self._channel_service.block_manager
+            if self._channel_service.state_machine.state == "BlockGenerate" and (
```

Comment:
Don't leaders block sync?

Reply:
Yes. Leader node change its type first and then do block sync next height.